jupyterlab/commands.py

```diff
@@ -784,14 +784,18 @@ def _update_extension(self, name):
 
         Returns `True` if a rebuild is recommended, `False` otherwise.
         """
+        data = self.info['extensions'][name]
+        if data["package_name"]:
+            self.logger.warn("Skipping updating pinned extension '%s'." % name)
+            return False
```

Comment:
This logic does not seem intuitive at all. Could we use a more descriptive field name than `package_name`? Ideally something which makes it clear that he presence of the filed implies this is an aliased/pinned package? E.g. `alias_package_source`?

Reply:
Sure, I can rename it to `alias_package_source`.